core/src/main/scala/org/http4s/EntityEncoder.scala

```diff
@@ -147,11 +147,23 @@ trait EntityEncoderInstances extends EntityEncoderInstances0 {
   implicit val byteArrayEncoder: EntityEncoder[Array[Byte]] =
     chunkEncoder.contramap(Chunk.bytes)
 
-  // TODO fs2 port this is where we miss ByteVector
-  /*
   implicit val byteBufferEncoder: EntityEncoder[ByteBuffer] =
-    chunkEncoder.contramap(ByteVector.view)
-   */
+    chunkEncoder.contramap { bb =>
+      val duplicate = bb.duplicate()
+      // This is triggered if puts have been done, then we want to flip to read
+      if (duplicate.remaining != duplicate.capacity()) {
+        duplicate.flip()
+      }
+      if (duplicate.hasArray) {
+        Chunk.bytes(duplicate.array, 0, duplicate.remaining)
+      } else {
+        // This will create a new copy of the array, maybe expensive
+        // Should we break this if the ByteBuffer is very large?
```

Comment:
I think in that case we'd have to do that as a `Process`, and lose the `contramap`.  We should still be able to provide a `Content-Length` header, unlike some of our other `Process` encoders.

I don't get deep into the NIO weeds very much.  In practice, do we see enormous non-array ByteBuffers in wild?